Dockerfile

```diff
@@ -1,38 +1,49 @@
 # BUILD IMAGE --------------------------------------------------------
-FROM alpine:3.11 AS nim-build
+
+FROM alpine:3.12 AS nim-build
 
 ARG NIM_PARAMS
 ARG MAKE_TARGET=wakunode
 
 # Get build tools and required header files
 RUN apk add --no-cache bash build-base pcre-dev linux-headers git
 
-RUN mkdir /app
 WORKDIR /app
-ADD . .
+COPY . .
 
 # Ran separately from 'make' to avoid re-doing
 RUN git submodule update --init --recursive
 
-# Build the node binary
-RUN make -j$(nproc) ${MAKE_TARGET} NIM_PARAMS="${NIM_PARAMS}"
+# Slowest build step for the sake of caching layers
+RUN make -j$(nproc) deps
+
+# Build the final node binary
+RUN make -j$(nproc) $MAKE_TARGET NIM_PARAMS="$NIM_PARAMS"
 
 # ACTUAL IMAGE -------------------------------------------------------
-FROM alpine:3.11
+
+FROM alpine:3.12
+
+ARG MAKE_TARGET=wakunode
 
 LABEL maintainer="jakub@status.im"
 LABEL source="https://github.com/status-im/nim-waku"
 LABEL description="Wakunode: Waku and Whisper client"
 
-EXPOSE 1234
+# LibP2P and JSON RPC ports
+EXPOSE 30303 8545
```

Comment:
I think the default libp2p port here is 60000


```
./build/wakunode2
INF 2020-09-22 10:00:44.031+08:00 Initializing networking                    topics="wakunode" tid=13489 file=wakunode2.nim:69 hostAddress=/ip4/0.0.0.0/tcp/60000 announcedAddresses=@[/ip4/192.168.1.101/tcp/60000]
```

Reply:
Not sure how it worked before with 1234 though, eh

Reply:
`EXPOSE` is just informational really. What matters is what ports you map when you create the container.

Reply:
I thought it was `30303` based on this:
https://github.com/status-im/nim-waku/blob/0bce0ba7eb5a0407bfea38d86c86272f249cee38/waku/node/v1/config.nim#L26-L34

Reply:
Oh, V2 uses different one:
https://github.com/status-im/nim-waku/blob/0bce0ba7eb5a0407bfea38d86c86272f249cee38/waku/node/v2/config.nim#L28-L36
So I guess we should `EXPOSE` both.

Reply:
Done.